### PR TITLE
Add support for building on macOS

### PR DESCRIPTION
**Pull request by jturner314-nrl**
This PR adds support for building Goby on macOS. The individual commit messages provide the details.

### CODE REVIEW

File: src/middleware/protobuf/coroner.proto [hunk 1]
```diff
@@ -23,7 +23,10 @@ enum Error
 message ThreadHealth
 {
     required string name = 1;
+    // Linux thread ID.
     optional int32 thread_id = 2;
+    // Apple thread ID.
+    optional uint64 thread_id_apple = 4;
```
Comment (jturner314-nrl) [inline src/middleware/protobuf/coroner.proto]: I'm somewhat unsatisfied with adding a separate `thread_id_apple` field to `ThreadHealth` here, but it seems like the cleanest solution.

Comment (tsaubergine) [reply src/middleware/protobuf/coroner.proto]: I made a small change to ThreadHealth that I think makes this cleaner:

```
message ThreadHealth
{
    required string name = 1;
    oneof t_id
    {
        // Linux thread ID.
        int32 thread_id = 2;
        // Apple thread ID.
        uint64 thread_id_apple = 4;
    }
    optional int32 uid = 3;
    required HealthState state = 10;
    repeated ThreadHealth child = 11;
    optional Error error = 20;
    optional string error_message = 21;

    extensions 1000 to max;
    // 1000 - jaiabot
}
```

**Comment by tsaubergine** [comment]
Could you please add your name to the AUTHORs file and commit to this branch? (This also accepts the Contributor License Agreemnt so I can use your contributions and release them as open source)

https://github.com/jturner314-nrl/goby3/blob/macos-support/AUTHORS

**Comment by jturner314-nrl** [comment]
I developed these changes as part of my official duties as a U.S. government employee, so they're in the public domain (can be used freely, not subject to U.S. copyright restrictions). See:

- [DoD Open Source Software FAQ, Question "Can government employees contribute code to open source software projects?"](https://dodcio.defense.gov/Open-Source-Software-FAQ/#q-can-government-employees-contribute-code-to-open-source-software-projects)
- [GPL FAQ, Question "Can the US Government release improvements to a GPL-covered program?"](http://www.gnu.org/licenses/gpl-faq.html#GPLUSGovAdd)

So, I pushed a commit to this PR which adds my name to the `AUTHORS` file in a separate section for U.S. government authors. Is that sufficient?

File: src/middleware/protobuf/coroner.proto (hunk 1 above)
Comment (jturner314-nrl) [reply src/middleware/protobuf/coroner.proto]: Yes, that's better. Thanks!

**Comment by jturner314-nrl** [comment]
The CI failures appear to be due to a change in DCCL, unrelated to my changes:

```
In file included from /usr/include/dccl/any.h:29,
                 from /usr/include/dccl/field_codec.h:35,
                 from /usr/include/dccl/codecs2/field_codec_default_message.h:28,
                 from /root/goby3/src/acomms/dccl/dccl.h:35,
                 from /root/goby3/src/acomms/dccl/dccl.cpp:25:
/usr/include/dccl/def.h:37:5: error: token "@" is not valid in preprocessor expressions
 #if DCCL_COMPILED_CXX_STANDARD >= 17
     ^~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/dccl/def.h:40:5: error: token "@" is not valid in preprocessor expressions
 #if DCCL_COMPILED_CXX_STANDARD >= 20
     ^~~~~~~~~~~~~~~~~~~~~~~~~~
```

**Comment by tsaubergine** [comment]
> I developed these changes as part of my official duties as a U.S. government employee, so they're in the public domain (can be used freely, not subject to U.S. copyright restrictions). See:
> 
>     * [DoD Open Source Software FAQ, Question "Can government employees contribute code to open source software projects?"](https://dodcio.defense.gov/Open-Source-Software-FAQ/#q-can-government-employees-contribute-code-to-open-source-software-projects)
> 
>     * [GPL FAQ, Question "Can the US Government release improvements to a GPL-covered program?"](http://www.gnu.org/licenses/gpl-faq.html#GPLUSGovAdd)
> 
> 
> So, I pushed a commit to this PR which adds my name to the `AUTHORS` file in a separate section for U.S. government authors. Is that sufficient?

Thank you that's great!

**Comment by tsaubergine** [comment]
> The CI failures appear to be due to a change in DCCL, unrelated to my changes:
> 
> ```
> In file included from /usr/include/dccl/any.h:29,
>                  from /usr/include/dccl/field_codec.h:35,
>                  from /usr/include/dccl/codecs2/field_codec_default_message.h:28,
>                  from /root/goby3/src/acomms/dccl/dccl.h:35,
>                  from /root/goby3/src/acomms/dccl/dccl.cpp:25:
> /usr/include/dccl/def.h:37:5: error: token "@" is not valid in preprocessor expressions
>  #if DCCL_COMPILED_CXX_STANDARD >= 17
>      ^~~~~~~~~~~~~~~~~~~~~~~~~~
> /usr/include/dccl/def.h:40:5: error: token "@" is not valid in preprocessor expressions
>  #if DCCL_COMPILED_CXX_STANDARD >= 20
>      ^~~~~~~~~~~~~~~~~~~~~~~~~~
> ```

Yes I'm breaking all kinds of things apparently with the forthcoming DCCL 4.1 changes.

**Pull request merged**